go/adbc/pkg/_tmpl/driver.go.tmpl

```diff
@@ -115,6 +198,44 @@ func getFromHandle[T any](ptr unsafe.Pointer) *T {
 	return cgo.Handle((uintptr)(*hptr)).Value().(*T)
 }
 
+func exportStringOption(val string, out *C.char, length *C.size_t) C.AdbcStatusCode {
+	lenWithTerminator := C.size_t(len(val) + 1)
+	if lenWithTerminator <= *length {
+		// TODO: doesn't this reallocate the string? can we avoid this?
+		C.strncpy(out, (*C.char)(C.CString(val)), lenWithTerminator)
+	}
+	*length = lenWithTerminator
+	return C.ADBC_STATUS_OK
+}
+
+func exportBytesOption(val []byte, out *C.uint8_t, length *C.size_t) C.AdbcStatusCode {
+	if C.size_t(len(val)) <= *length {
+		sink := fromCArr[byte]((*byte)(out), int(*length))
+		copy(sink, val)
+	}
```

Comment:
should we throw some error if the buffer they provided isn't large enough? or should it just truncate (in which case you don't need the length check since `copy` will already truncate if the source is larger than the destination)

Reply:
No, there's no point. The definition in adbc.h describes how to handle this.